### PR TITLE
add CSGManager to editor persistency in IndexedDB (#72)

### LINKED ISSUE #72: save state of created geometry (zones and bodies)
it should be persitent on refresh

### PR DESCRIPTION
**Pull request by ostatni5**
Add to the existing layer of persistency state of the CSGManager.

Remember to wait until DB saves state.
Example log:
obraz

Closes #72.

### CODE REVIEW

File: src/ThreeEditor/util/CSGManager.ts [hunk 1]
```diff
@@ -1,34 +1,75 @@
...
+
+    toJSON() {
+        let jsonObject: CSGOperationJSON = {
+            mode: this.mode,
+            objectUuid: this.object.uuid
+        };
+        return jsonObject;
+    }
+
+    static formJSON(editor: Editor, data: CSGOperationJSON) {
```
Comment (grzanka) [inline src/ThreeEditor/util/CSGManager.ts:33]: Should it be form or from ?

Comment (ostatni5) [reply src/ThreeEditor/util/CSGManager.ts:33]: Good catch, `from` is correct

**Review by grzanka — DISMISSED** [review]
works well

**Review by grzanka — APPROVED** [review]
ok

File: src/ThreeEditor/js/Editor.js [hunk 2]
```diff
@@ -91,6 +92,10 @@ function Editor() {
 		layoutChanged: new Signal(), // Layout signal 
 
 		viewportConfigChanged: new Signal(), // Viewport config signal 
+		
+		CSGManagerStateChanged: new Signal(), // State of CSGmanager changed
```
Comment (Derstilon) [inline src/ThreeEditor/js/Editor.js:96]: We should consider signal naming that can start with lower case

**Review by Derstilon — APPROVED** [review]
See the [comment ](https://github.com/yaptide/ui/pull/80/files#r709469088) I've made.

**Pull request merged**